src/js/game/upgrades.js

```diff
@@ -22,8 +29,13 @@ function generateEndgameUpgrades() {
     }));
 }
 
-for (let i = 0; i < numEndgameUpgrades; ++i) {
-    fixedImprovements.push(0.1);
+if (numEndgameUpgrades != 0) {
+    for (let i = 0; i < 8; i++) {
+        fixedImprovements.push(0.5);
+    }
+    for (let i = 0; i < 4; i++) {

```

Comment:
```suggestion
    for (let i = 0; i < numEndgameUpgrades - 4; i++) {

```
This should make this work even if the number of upgrades changes